doc/pmon/pmon-sensormon.md

```diff
@@ -0,0 +1,219 @@
+
+Note that temperature sensor devices are managed via SONiC ThermalCtlD daemon today. At this point there is no change proposed for ThermalCtlD. This proposed design can be used for voltage, current and other types of sensors.
+
+Linux does provide some support of voltage and current sensor monitoring using lmsensors/hwmon infrastructure. However there are a few limitations with that
+
+- Devices not supported with Hwmon are not covered
+- Simple devices which donot have an inbuilt monitoring functions do not generate any alarms
+- Platform specific thresholds for monitoring are not available
+
+The solution proposed in this document tries to address these limitations by extending the coverage to a larger set of devices and providing platform specific thresholds for sensor monitoring.
+
+
+### Requirements
+
+This HLD covers
+
+* Discovery of voltage and current sensor devices in the system
+* Monitoring the sensor devices periodically and update that data in Redis DB
+* Raise/Clear alarms if the sensor devices indicate readings which are unexpected and clear them if they return to a good state.
+* Report sensor alarm conditions in system health
+* Enable such sensors in Entity MIB
+* A framework for adding new sensor types in future
+
+This HLD does not cover
+
+* An automated recovery action that system might take as a result of a fault reported by the voltage sensor device. A network management system may process the alarms and take recvoery action as it sees fit.
+
+
+### High Level Design 
+
+The proposal for monitoring sensor devices is to create a new SensorMon daemon. SensorMon will use API provided by platform to discover the sensor devices. It will periodically poll the devices to refresh the sensor information and update the readings in StateDB.
+
+If the sensor device readings cross the minor/major/critical thresholds, syslogs will be generated to indicate to the operator about the alarm condition. If the sensor reports normal data in a subsequent poll, another syslog will be generated to indicate that the fault condition is cleared.
+
+CLIs are provided to display the sensor devices, their measurements, threshold values and if they are reporting an alarm.
+
+Platform APIs will provide 
+
+* List of sensors devices of a specific type in the system
+* Way to read the sensor information
+
+The following SONiC repositories will have changes
+
+#### sonic-platform-daemons	
+
+SensorMon will be a new daemon that will run in PMON container. It will retrieve a list of sensors of different sensor types from the platform during initialization. Subsequently, it will poll the sensor devices on a periodic basis and update their measurments in StateDb. SensorMon will also raise syslogs on alarm conditions. 
+
+Following is the DB schema for voltage and current sensor data.
+
+##### Voltage Sensor StateDb Schema
```

Comment:
@bmridul in which format platform specify different onboard sensors and respective threshold Can you make the format public?

Reply:
@prgeor , Pls check 
https://github.com/sonic-net/SONiC/blob/3ed3cca2866e3647b8af9871b33dc53b5dbae1a6/doc/pmon/pmon-sensormon.md#platform-sensors-configuration

Reply:
@bmridul so this will work only for platforms having sysfs for each sensors?

Reply:
Yes,  For platforms having sysfs based sensors, they only need to provide sensors details in this file and Sensormon will use that.